### ISSUE TITLE
Relative paths to `node_modules` being reported as unpublished

### ISSUE THREAD

**Issue by brettz9** [mysticatea/eslint-plugin-node#162]
Hi,

I'm getting `no-unpublished-import` errors with `node_modules`-based references:

```js
import Typeson from '../node_modules/typeson-registry/dist/index.js';
```

This use of paths is convenient in the browser for examples/demos where one doesn't wish to go through a build routine and doesn't need to support older browsers.

**Comment by mysticatea** [comment]
Thank you for this report.

However, this is intended because `npm` can hoist installed packages and break such a path.

**Comment by brettz9** [comment]
If it is an application (rather than a library), I would think this should not be an issue...

**Comment by mysticatea** [comment]
In that case, you can disable the rule in your application's repo.

**Comment by brettz9** [comment]
Might you consider allowing this behavior through an option? The rule has been quite helpful, but for demos/testing I like to use the `node_modules` paths directly and avoid the build steps.

**Comment by mysticatea** [comment]
Hmm. I'm not sure about the usecase...

To make clear, you have to include the demos/testing files in `npm publish`?

**Comment by mysticatea** [comment]
Also, `npm` may install that dependency into `./node_modules/typeson-registry/dist/index.js` rather than `../node_modules/typeson-registry/dist/index.js` if there is the version conflict of `typeson-registry`. Is this not a problem?

**Comment by brettz9** [comment]
I sometimes build simple applications and distribute them through `npm publish`, and it depends on `node_modules` content, but I prefer to work without a build step slowing my debugging down (or dealing with "watch" routines which occur too slowly for my frequent changes).

**Comment by brettz9** [comment]
I recognize that the situation you mention could be a problem in some cases, but I'd prefer to use the coding approach regardless (and I tend to be familiar enough with my dependencies to know about any such conflicts).

**Comment by mysticatea** [comment]
Thank you for your answer.

I'm not a fan of depending on the implementation details of a package manager because I have encountered several problems about that on `eslint` repo. However, I understand browsers supported ESM natively and it's useful for applications.

I'm fine for adding the option to allow relative paths.
